README.md

```diff
@@ -1,10 +1,153 @@
-# hack-game-state-service
+# GameStateService
 
-## Development
+Simply put, it's the player's save-file. More precisely, is a central service that manages all the metadata related to the user's progression in Hack, and is available to all components of the desktop. Technically speaking:
+
+1. Manages all reading and writing operations to the player's state-file.
+2. Provides a DBus interface that allows other components of the desktop to interact with the player's state.
+
+## Player's State File
+
+The state-file is just a plain-text file. Progression metadata is stored in this file using the JSON format, in order to keep it human-readable. By default, this file will be located at:
+
+```
+~/.var/app/com.endlessm.GameStateService/data/state.json
+```
+
+As mentioned above, the contents of this `state.json` file are objects serialized with the JSON format. See the following example:
+
+```json
+{
+  "item.key.fizzics.1" : {
+    "consume_after_use" : true,
+    "used" : true
+  },
+  "lock.fizzics.1 : {
+    "locked" : true
+  }
+}
+```
+
+The example above shows entries that were written and read by the [Clubhouse](https://github.com/endlessm/clubhouse) and by the [Toolbox](https://github.com/endlessm/hack-toolbox-app). In the example, the Clubhouse has given the player the first key in the Fizzics quest line, which was later used and updated by the Toolbox to unlock the first Fizzics lockscreen. All that interaction between the Clubhouse, the Toolbox and the player's state is done throught the GameStateService.
+
+**NOTE**: the root object of player's state must be a dictionary, while the sub-objects can, in theory, use any kind of serializable objects. When in doubt, make sure to test that your objects are being serialized properly, by manually reading the state file.
+
+**NOTE**: the GameStateService does not provide, nor enforces, any kind content schema or structure. The contents of the state-file are entirely defined by the other components of the desktop. Therefore, it requires an agreement between these components to honour these definitions. For now, it is the price of simplicity.
+
+## The DBus Service
+
+The GameStateService exposes the API consumed by other components of the desktop to access and manipulate the player's state.
+
+### Writing To The State
+
+```javascript
+const {Gio} = imports.gi;
+
+const BusName = 'com.endlessm.GameStateService';
+const BusPath = '/com/endlessm/GameStateService';
+const BusIface = `
+<node>
+  <interface name='com.endlessm.GameStateService'>
+    <method name='Set'>
+      <arg type='s' name='key' direction='in'/>
+      <arg type='v' name='value' direction='in'/>
+    </method>
+  </interface>
+</node>
+`;
+const GameStateProxy = Gio.DBusProxy.makeProxyWrapper(BusIface);
+const gameStateProxy = new GameStateProxy(Gio.DBus.session, BusName, BusPath);
+
+const variant = new GLib.Variant('a{sb}', {locked: false});
```

Comment:
The clubhouse builds the variant using `json-glib`. This way we don't have to write the signature, `b` for boolean for example. I think is easier:
```
                json_str = json.dumps(obj)
                variant = Json.gvariant_deserialize_data(json_str, -1, None)
```
Where `obj` is a python object that can be serialized into JSON.